packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -146,29 +164,30 @@ export const calculateImageSizes = ({
 
     // Create sizes (in width) for the image if no custom breakpoints are
     // provided. If the max width of the container for the rendered markdown file
-    // is 800px, the sizes would then be: 200, 400, 800, 1200, 1600.
+    // is 800px, the sizes would then be: 200, 400, 800, 1600, 2400 (based off of
+    // the default outputPixelDensities)
     //
     // This is enough sizes to provide close to the optimal image size for every
     // device size / screen resolution while (hopefully) not requiring too much
     // image processing time (Sharp has optimizations thankfully for creating
     // multiple sizes of the same input file)
 
-    // use standard breakpoints if no custom breakpoints are specified
-    if (!srcSetBreakpoints || !srcSetBreakpoints.length) {
-      sizes = densities
-        .map(density => density * width)
-        .filter(size => size <= maxWidth)
-      if (!sizes.includes(maxWidth)) {
-        sizes.push(maxWidth)
-      }
-      sizes = sizes.sort()
-    } else {
-      let sizes = srcSetBreakpoints.filter(size => size <= maxWidth)
-      if (!sizes.includes(maxWidth)) {
-        sizes.push(maxWidth)
-      }
-      sizes = sizes.sort()
+    sizes = densities.map(density => density * fixedValue)
+    // add breakpoint set to sizes if they are specified
+    if (srcSetBreakpoints) {
```

Comment:
srcSetBreakpoints should override everything else

Reply:
Okay, in that case, do you think we should warn that any provided outputPixelDensities would have no effect if both are passed in? `srcSetBreakpoints` and `outputPixelDensities` approach the same kind of thing (custom specified sizes) from different angles. It seems a little confusing how they'd work together in that case.

The current resizing function adds the breakpoints to the array, but I can make it use them exclusively.

Reply:
Yeah, I think it should be one or the other. A warning is a good idea. What do you reckon, @laurieontech ?

Reply:
Ya, people might miss it, but always good to let them know when their code is ignored

Reply:
Okay, I'll add a warning then, thanks! My suspicion is _most_ people won't bother to use either one 😅 but if they need to, it should be clear.

Reply:
Added a warning [in a new commit](https://github.com/gatsbyjs/gatsby/pull/27554/commits/e4215704e77b6b8190dc7fd423d18815ab68b52e#diff-36db1668fae19fc6f3b88538a610293a66071255a4633e383b76b9b7e79b97edR203-R207), and updated the logic so only `srcSetBreakpoints` will be used when they are passed in.